pkg/apis/eventing/v1alpha1/channel_types_test.go

```diff
@@ -101,6 +101,9 @@ func TestChannelInitializeConditions(t *testing.T) {
 			}, {
 				Type:   ChannelConditionProvisioned,
 				Status: corev1.ConditionUnknown,
+			}, {
+				Type:   ChannelConditionProvisionerInstalled,
+				Status: corev1.ConditionUnknown,
```

Comment:
This is an example of what I am saying in the ` InitializeConditions() ` comment. This should be `True` because the only thing that should Init the conditions is the real controller.

Reply:
> There is also a question of if this controller should update this status on the normal re-reconcile period if it finds there to be no provisioner. It would sure be interesting to get that set on the channel that worked and then the provisioner went away and now my stuff is broken?

The scenario is valid but as per our previous conversation we think it is not likely. The only solution we discussed was to update the the Channel in case it wasn't updated by anyone for more than 8hr (which is the default time for reconciliation in case there are no updates). This in itself is a long delay for the feedback and doesn't help much. It requires more thinking and we can get to it if we see it happening or based on user feedback.